Makefile

```diff
@@ -3,13 +3,76 @@
 # These variables can be overridden by setting an environment variable.
 LOCAL_IP_ENV?=$(shell ip route get 8.8.8.8 | head -1 | cut -d' ' -f8)
 
+CONFD_REPO?=calico/confd
+CONFD_VER?=v1.0.0-beta1-4-g4619952
+CONFD_CONTAINER_NAME=${CONFD_REPO}:${CONFD_VER}
+
 default: clean calicorr.created
 
-calicorr.created: $(BUILD_FILES)
+calicorr.created: $(BUILD_FILES) dist/confd
 	docker build -t calico/routereflector .
 	touch calicorr.created
 
 clean:
 	-rm *.created
-	-docker rm -f calico-rr
-	-docker rmi -f calico/rr
+	-rm -rf dist
+	-docker rmi -f calico/routereflector
+	-docker rmi -f quay.io/calico/routereflector
+
+dist/confd: dist
+	-docker rm -f calico-confd
+	# Latest confd binaries are stored in automated builds of calico/confd.
+	# To get them, we create (but don't start) a container from that image.
+	docker pull $(CONFD_CONTAINER_NAME)
+	docker create --name calico-confd $(CONFD_CONTAINER_NAME)
+	# Then we copy the files out of the container.  Since docker preserves
+	# mtimes on its copy, check the file really did appear, then touch it
+	# to make sure that downstream targets get rebuilt.
+	docker cp calico-confd:/bin/confd $@ && test -e $@ && touch $@
+	-docker rm -f calico-confd
+	chmod +x $@
+
+dist:
+	mkdir -p dist
+
+release: clean
+ifndef VERSION
+	$(error VERSION is undefined - run using make release VERSION=vX.Y.Z)
+endif
+	git tag $(VERSION)
+
+	# Check to make sure the tag isn't "-dirty".
```

Comment:
I think you can move this check above the `git tag` if you do `git describe --always --dirty`